completion/available/sdkman.completion.bash

```diff
@@ -8,24 +8,24 @@ function _sdkman_complete() {
 	COMPREPLY=()
 
 	if [ "$COMP_CWORD" -eq 1 ]; then
-		mapfile -t COMPREPLY < <(compgen -W "install uninstall rm list ls use default home env current upgrade ug version broadcast help offline selfupdate update flush" -- "${COMP_WORDS[COMP_CWORD]}")
+		while IFS='' read -r line; do COMPREPLY+=("$line"); done < <(compgen -W "install uninstall rm list ls use default home env current upgrade ug version broadcast help offline selfupdate update flush" -- "${COMP_WORDS[COMP_CWORD]}")
 	elif [ "$COMP_CWORD" -eq 2 ]; then
 		case "${COMP_WORDS[COMP_CWORD - 1]}" in
 			"install" | "i" | "uninstall" | "rm" | "list" | "ls" | "use" | "u" | "default" | "d" | "home" | "h" | "current" | "c" | "upgrade" | "ug")
 				CANDIDATES="${SDKMAN_CANDIDATES_CSV//,/${IFS:0:1}}"
-				mapfile -t COMPREPLY < <(compgen -W "$CANDIDATES" -- "${COMP_WORDS[COMP_CWORD]}")
+				while IFS='' read -r line; do COMPREPLY+=("$line"); done < <(compgen -W "$CANDIDATES" -- "${COMP_WORDS[COMP_CWORD]}")
```

Comment:
Even if we wouldn't handle this in an ideal but more involved way for now, the repetition of `while IFS='' read -r line; do COMPREPLY+=("$line"); done < <(compgen ...)` seems ugly. Couldn't a function such as `_store_compgen` be defined in e.g. `lib/helpers.bash` to use it everywhere it is available? Define something like

```bash
_store_compgen() {
	local line
	while IFS='' read -r line; do COMPREPLY+=("$line"); done < <(compgen "$@")
}
```

and use it as e.g.

```suggestion
				_store_compgen -W "$CANDIDATES" -- "${COMP_WORDS[COMP_CWORD]}"
```

Reply:
Even if I do that, it may solve it for most of the mapfile cases, but there are still a few that are not compgen -> COMPREPLY

I feel like it's a problem for another day. it's not a code cleanup job, it's a bit of a re-factor. separate PR.

Reply:
Then, I suggest keeping the current version instead of rewriting it to `while IFS='' ... `. To me rewriting it to `while IFS='' ...` seems also a refactoring (or a fix for the problem of only picking the first line), which is unrelated to the current cleanup PR. It should also be noted that [most of `while IFS='' ... ` were `COMPREPLY=($(compgen ...))`](https://github.com/Bash-it/bash-it/pull/2265/files#diff-e232dd26c7e604482c2665680825011f79ac60c5c9ad0cdf345435576a330592) before #2265, so the current change isn't even the reverting.

Reply:
well, shellcheck is unhappy with `COMPREPLY=($(compgen ...))`, so I stil consider it a cleanup rather than a refactor. I prefer not to flood code with `#shellcheck disable` lines more than is absolutely needed, does that make sense?

Reply:
OK, thanks for sharing your thoughts. I admit that there is no perfect solution here, so let's go with the current version. The refactoring could be done later.